### PR TITLE
 split the crate in 2 separate crates 

### PR DESCRIPTION
**Pull request by antoine-de**
This has been long overdue.

This split cosmogony into 2 crates:

- cosmogony with all the cosmogony types, that will be pushed to crates.io
- cosmogony_builder: only used by the cosmogony binary, used to create the hierachy.

In addition to being cleaner, this remove the dependency on libgeos for `cosmogony_model` :tada:

As a side addition I also cleanup the `disable_geom` parameter, as it is now completely useless.

Note: I haven't done the travis yet, I first want your thoughts on this, If the split is ok for you, I think we can do an automatic deployment only for `cosmogony`,

### CODE REVIEW

**Comment by amatissart** [comment]
It looks good to me :+1: 

So Travis would only publish a new "cosmogony_model" crate, right ?

**Comment by antoine-de** [comment]
Yes I think we only need this crate, I don't plan to release `cosmogony` anymore.

Do you know a way to thus mark `cosmogony` deprecated in crates.io ?

I don't think I can call the new sub crate `cosmogony` :cry:

**Comment by antoine-de** [comment]
ok, I renamed `cosmogony_model` to `cosmogony` and `cosmogony` (the root project) to `cosmogony_builder`

this way the users can only update the version of cosmogony

File: Cargo.toml [hunk 1]
```diff
@@ -1,28 +1,21 @@
 [package]
-name = "cosmogony"
-version = "0.5.3"
+name = "cosmogony_builder"
+version = "0.6.0"
```
Comment (GuillaumeGomez) [inline Cargo.toml]: Just to underline: it starts right off at `0.6` version. However, it sounds like a good idea to match `cosmogony_builder` version with `cosmogony`'s.

File: cosmogony/src/file_format.rs [hunk 2]
```diff
@@ -16,10 +17,16 @@ static ALL_EXTENTIONS: [(&str, OutputFormat); 4] = [
...
+            .find(|&&(ref e, _)| {
+                filename
+                    .as_ref()
+                    .file_name()
+                    .and_then(|f| f.to_str())
+                    .map_or(false, |f| f.ends_with(e))
+            })
             .map(|&(_, ref f)| f.clone())
             .ok_or_else(|| {
                 let extensions_str = ALL_EXTENTIONS
```
Comment (GuillaumeGomez) [inline cosmogony/src/file_format.rs]: `ALL_EXTENTIONS` -> nice typo :)

**Comment by antoine-de** [comment]
oops good catch for the .dockerignore and the typo, corrected :wink:

**Pull request merged**